code/script.js

```diff
@@ -218,108 +220,146 @@ const generateBoard = () => {
         </div>
       </div>
-    `
-  })
-}
+    `;
+  });
+};
 
-// Randomly select a person from the characters array and set as the value of the variable called secret
+// Randomly select a person from the characters array
+// Set selected person as the secret character
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+  secret =
+    charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)];
+};
 
-// This function to start (and restart) the game
+// Start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  // Set charactersInPlay array to include all the characters we start with
+  charactersInPlay = CHARACTERS;
+  // Invoke the function that draws the game board
+  generateBoard();
+  // Pick a secret character
+  setSecret();
+  // Hide the win or lose screen
+  winOrLoseSection.style.display = "none";
+};
 
-// setting the currentQuestion object when you select something in the dropdown
+// Select a question in the dropdown and set its value to currentQuestion
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  // Store what "option group" (category) the question belongs to
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  // Store the actual value of the selected question
+  const value = questions.value;
+  // Store category and value in the current question
   currentQuestion = {
-    category: category,
-    // value: value
-  }
-}
+    category: category, // Based on the optgroup
+    value: value, // Comes from the selected option
+  };
+};
 
-// This function should be invoked when you click on 'Find Out' button.
+// After clicking on the 'Find Out' button, compare the details from currentQuestion with the secret person
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  const { category, value } = currentQuestion;
+  // Comparison for key pairs with string values
+  if (category === "hair" || category === "eyes") {
+    if (secret[category] === value) {
+      filterCharacters(true);
+    } else {
+      filterCharacters();
```

Comment:
In one of our group conversations, people were discussing if we should write "false" or leave the boolean empty. I found an article that really helped me understand this: 
https://medium.com/swlh/avoid-null-booleans-in-java-4a5cd9b23bca

Reply:
Oh wow, that's a really useful article! I myself wasn't so sure about it, but now it's clear to me. Thanks a lot Bruna!